python/nano/tutorial/notebook/training/pytorch/accelerate_pytorch_training_bf16.ipynb

```diff
@@ -0,0 +1,382 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use BFloat16 Mixed Precision for PyTorch Training"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Brain Floating Point Format (BFloat16) is a custom 16-bit floating point format designed for machine learning. BFloat16 is comprised of 1 sign bit, 8 exponent bits, and 7 mantissa bits. With the same number of exponent bits, BFloat16 has the same dynamic range as FP32, but requires only half the memory usage.\n",
+    "\n",
+    "BFloat16 Mixed Precision combines BFloat16 and FP32 during training, which could lead to increased performance and reduced memory usage. Compared to FP16 mixed precision, BFloat16 mixed precision has better numerical stability.\n",
+    "\n",
+    "By using `TorchNano` (`bigdl.nano.pytorch.TorchNano`), you can make very few code changes to use BFloat16 mixed precision for training. Here we provide __2__ ways to achieve this: A) subclass `TorchNano` or B) use `@nano` decorator. You can choose the appropriate one depending on your (preferred) code structure."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "source": [
+    "## Prepare Environment for BigDL-Nano"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "source": [
+    "At first, you need to install BigDL-Nano for PyTorch:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install --pre --upgrade bigdl-nano[pytorch] # install the nightly-built version\n",
+    "!source bigdl-nano-init # set environment variables"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> 📝 **Note**\n",
+    ">\n",
+    "> Before starting your PyTorch application, it is highly recommended to run `source bigdl-nano-init` to set several environment variables based on your current hardware. Empirically, these variables will greatly improve performance for most PyTorch applications on training workloads."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "source": [
+    "> ⚠️ **Warning**\n",
+    "> \n",
+    "> For Jupyter Notebook users, we recommend to run the commands above, especially `source bigdl-nano-init` before jupyter kernel is started, or some of the optimizations may not take effect."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> ⚠️ **Warning**\n",
+    "> \n",
+    "> Using BFloat16 precision with `torch < 1.12` may result in extremely slow training."
```

Comment:
running bf16 on CPU without bf16 instruction support is also slow.

Reply:
From my understanding, running bf16 is not supported on CPU without bf16 instruction sets, either with or without IPEX. Maybe we should put this sentence in the warning box here?
![image](https://user-images.githubusercontent.com/54161268/210318015-df1b19c6-1d8c-45ce-af48-d8d84ac0552a.png)

For me, the sentence here sounds like that bf16 with IPEX can be a work around for CPU without bf16 instruction sets, which seems to be not correct.

Reply:
`bf16` requires bf16 instruction, such as `avx512_bf16`, while `IPEX+bf16` only requires `avx512` instruction (`avx512` doesn't belong to bf16 instruction), so `IPEX+bf16` is indeed a workaround.